developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -400,7 +400,158 @@ response.data?.Get?.Article?.[0]['_additional']?.creationTimeUnix // Get the tim
 
 ## How to migrate your code
 
-For code examples, see the pages here:
+
+This workflow describes the process one would follow to migrate their codebase to the v3 client. 
+
+1. Install the v3 client package. Update the client package from the v2 to v3 client. 
+
+2. Connect to Weaviate. Make sure your client instantiation respects the new clients syntax. Our documentation on instantiation covers use cases from local, custom to WCS.
+
+3. Migrate your code snippets. The v3 client is collection first which means we do not have to duplicate mentions of the collection you are interacting with.
+
+Below is an example of a basic script in both the v2 and v3 clients. In this script, we define a collection, insert then query objects from the collection, and finally delete the collection. This is to give you an idea of what the migration would entail.
+
+<Tabs groupId="languages">
+<TabItem value="jsv3" label="JS/TS (v3)">
+
+```ts
+import weaviate from 'weaviate-client'
+require('dotenv').config();
+
+async function main() {
+
+  const collectionName = 'Recipes'
+
+  const client = await weaviate.connectToWCS(
+    process.env.WEAVIATE_URL || '',
+    {
+      authCredentials: new weaviate.ApiKey(process.env.WEAVIATE_API_KEY || ''),
+      headers: {
+        'X-OpenAI-Api-Key': process.env.OPENAI_API_KEY || '',  
+      }
+    }
+  )
+
+  await client.collections.create({
+    name: collectionName,
+    vectorizer: weaviate.configure.vectorizer.text2VecOpenAI(),
+  })
+
+  const recipeList = [
+    {"Name": "Christmas pie", "Description": "Combine a few key Christmas flavours here to make a pie that both children and adults will adore", "Author": "Mary Cadogan", "Ingredients": ["2 tbsp olive oil", "knob butter", "1 onion, finely chopped", "500g sausagemeat or skinned sausages", "grated zest of 1 lemon", "100g fresh white breadcrumbs", "85g ready-to-eat dried apricots, chopped", "50g chestnut, canned or vacuum-packed, chopped", "2 tsp chopped fresh or 1tsp dried thyme", "100g cranberries, fresh or frozen", "500g boneless, skinless chicken breasts", "500g pack ready-made shortcrust pastry", "beaten egg, to glaze"]},
```

Comment:
This sample data is really long. Is it different in v2 and v3? 

If it is the same, can we import it instead so the other changes are more visible. 

If it is different can we parse it down to a minimum example? Maybe only 2 recipes?

Reply:
I think we could use an import. It's the same sample data.Also happy to use data from somewhere else if we have a more standard dataset.